src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_load_store_emitters.cpp

```diff
@@ -25,6 +25,89 @@ namespace ov::intel_cpu::aarch64 {
 using jit_generator = dnnl::impl::cpu::aarch64::jit_generator;
 using cpu_isa_t = dnnl::impl::cpu::aarch64::cpu_isa_t;
 
+// Helper function to load with large offset handling
+template <typename RegType>
+static void load_with_offset_check(jit_generator* h, const RegType& dst, const XReg& src, int offset) {
+    if constexpr (std::is_same_v<RegType, VReg> || std::is_same_v<RegType, QReg>) {
+        // Manual offset handling for VReg/QReg due to uni_ldr limitations
+        const int off_mod = offset % 16;
+        const int off_mul_vl = offset / 16;
+
+        if (off_mod == 0 && off_mul_vl >= 0 && off_mul_vl <= 4095) {
+            h->ldr(QReg(dst.getIdx()), ptr(src, offset));
```

Comment:
Should we cast `offset` to `uint32_t` here too?

The same comment is for `Store` - please align `load` and `store` functions

Reply:
Yes, done